__docs__/wonder-blocks-card/card.stories.tsx

```diff
@@ -0,0 +1,306 @@
+    },
+    render: () => (
+        <Card styles={{root: styles.card}}>
+            <View style={[styles.gemRow, styles.gemHeaderRow]}>
+                <View style={styles.gemColumn}>
+                    <View style={styles.gemRow}>
+                        <Heading tag="h3" size="large">
+                            Gem challenge
+                        </Heading>
+                        <PhosphorIcon
+                            icon={infoIcon}
+                            size="small"
+                            color={semanticColor.feedback.info.strong.icon}
+                        />
+                    </View>
+                    <GemBadge label="30 days left" showIcon={false} />
+                </View>
+                <View style={styles.gemIcon}>
+                    <Icon size="large">
+                        <GemIcon aria-label="Gem" />
+                    </Icon>
+                </View>
+            </View>
+            <View style={styles.gemColumn}>
+                {/* progress bar here */}
+                <View style={styles.gemRow}>
+                    <BodyText size="small" style={{marginInlineEnd: "auto"}}>
+                        Class gems
+                    </BodyText>
+                    <BodyText>
+                        <span>0</span> of <span>1500</span>
+                    </BodyText>
+                </View>
+                <View style={styles.gemRow}>
+                    <BodyText size="small" style={{marginInlineEnd: "auto"}}>
+                        Remaining
+                    </BodyText>
+                    <BodyText>
+                        <span>1500</span>
+                    </BodyText>
+                </View>
+                <View style={styles.gemRow}>
+                    <BodyText size="small" style={{marginInlineEnd: "auto"}}>
+                        Reward
+                    </BodyText>
+                    {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+                    <Link href="#" style={{fontSize: font.body.size.small}}>
+                        Set Reward
+                    </Link>
+                </View>
+                <Button kind="secondary">Learn more about gems</Button>
+                <Button kind="tertiary">View previous challenges</Button>
+            </View>
+        </Card>
+    ),
+};
+
+/**
+ * Cards also have the option to display a "close" button that can dismiss the card from the DOM.
+ */
+export const WithDismissButton: StoryComponentType = {
+    args: {
+        onDismiss: () => {},
+        labels: {dismissButtonAriaLabel: "Dismiss"},
+    },
+    render: (args) => {
+        const CardWithRef = () => {
+            const wrapperRef = React.useRef<HTMLDivElement>(null);
+            const cardRef = React.useRef<HTMLDivElement>(null);
+            const onDismiss = () => {
+                // Remove the card from the DOM
+                cardRef.current?.remove();
```

Comment:
The card doesn't get removed for me when I click the close button! https://5e1bf4b385e3fb0020b7073c-macgfgvdbn.chromatic.com/?path=/story/packages-card-card--with-dismiss-button

As an alternative to removing the element from the DOM, maybe we could use state to not render the card once the close button is pressed!

Reply:
Ah, it was an issue with `args` conflicting with the inline props. I cleaned those up and it works now!